### PR TITLE
resolve the crash on gridTemplateRows GridTemplateColumns missing

### PR DESCRIPTION
**Pull request by siddharthapaul001**
If gridTemplateColumns and gridTemplateRows attribute is missing then traverse over the children and find the maximum required tracklines and fill the sizeList array with 'auto' for each tracks. And modified the code for visualiser html and css-mason-compartor.js to visualize the results.

- added 'none' to validSIzes as 'none' should be the default value of gridTemplateColumns and gridTemplateRows according to [here](https://www.w3schools.com/cssref/pr_grid-template-columns.asp)
- added getMaxRowColumn function to determine required number of tracksizes in line 72
- added some code to fill sizeList with 'auto' for required number of tracks when gridTemplateRows or gridTemplateColumns are missing or given value 'none' in line 183 as 
> Columns are created if needed

### CODE REVIEW

File: src/grid/index.js [hunk 1]
```diff
@@ -164,15 +179,25 @@ class Grid {
       return false;
     }).map(size => getCleanSize(size));
 
-    for (i = 0, len = sizeList.length; i < len; i++) {
+    len = sizeList.length;
+    if(tracks === "none"){
+      len = this.getProps("maxTracks");
+      console.log(len);
```
Comment (n0x3u5) [inline src/grid/index.js]: @siddharthapaul001 This console.log statement should be removed.

@priyanjitdey94 We should probably add a lint rule to check for there errors.

Comment (priyanjitdey94) [inline src/grid/index.js]: remove this console.log statement

File: src/grid/index.js [hunk 2]
```diff
@@ -164,15 +179,25 @@ class Grid {
...
+      sizeList = 'auto,'.repeat(len).split(",");
```
Comment (n0x3u5) [inline src/grid/index.js]: @siddharthapaul001 Mason is intended to be used even in IE8. `String#repeat` is not supported in any version on IE at all.

File: src/grid/index.js [hunk 3]
```diff
@@ -164,15 +179,25 @@ class Grid {
...
+      sizeList.pop();
+    }
+
+    for (i = 0; i < len; i++) {
       startLineNames = (nameList[i] && nameList[i].replace(/\[|\]/g, '').split(' ').filter(name => name.length).map(name => name.trim())) || [i + 1 + ''];
       endLineNames = (nameList[i + 1] && nameList[i + 1].replace(/\[|\]/g, '').split(' ').filter(name => name.length).map(name => name.trim())) || [i + 2 + ''];
 
+      //getMaxRowColumn();
```
Comment (n0x3u5) [inline src/grid/index.js]: @siddharthapaul001 If this line is not required any more, please remove this from the final PR.

File: src/grid/index.js [hunk 4]
```diff
@@ -164,15 +179,25 @@ class Grid {
...
       sanitizedTracks.push({
         start: i + 1,
         end: i + 2,
-        size: sizeList[i],
+        size: sizeList[i]
       });
+      //console.log(sanitizedTracks);
```
Comment (n0x3u5) [inline src/grid/index.js]: @siddharthapaul001 If this line is not required any more, please remove this from the final PR.

File: tests/utils.js [hunk 5]
```diff
@@ -51,7 +51,110 @@ const basicTwoCrossTwoNodeWithFourChildren = {
   ]
 };
 
+const basicTestgridTemplateRowsColumnsNotProvided2 = {
```
Comment (n0x3u5) [inline tests/utils.js]: @siddharthapaul001 Please try and provide a more relevant name to the variable.

File: tests/utils.js [hunk 6]
```diff
@@ -51,7 +51,110 @@ const basicTwoCrossTwoNodeWithFourChildren = {
...
+        gridRowStart: 'two',
+        gridRowEnd: 'three',
+        gridColumnStart: 'two',
+        gridColumnEnd: 'three'
+      }
+    }
+  ]
+};
+
+const basicTestgridTemplateRowsColumnsNotProvided = {
```
Comment (n0x3u5) [inline tests/utils.js]: @siddharthapaul001 Please try and provide a more relevant name to the variable.

File: src/grid/index.js (hunk 1 above)
Comment (siddharthapaul001) [reply src/grid/index.js]: Done

**Review by n0x3u5 — CHANGES_REQUESTED** [review]

File: src/grid/index.js (hunk 2 above)
Comment (siddharthapaul001) [reply src/grid/index.js]: Used another alternative.

File: tests/utils.js (hunk 5 above)
Comment (siddharthapaul001) [reply tests/utils.js]: renamed it

File: tests/utils.js (hunk 6 above)
Comment (siddharthapaul001) [reply tests/utils.js]: Yes I renamed it in latest commit.

**Review by priyanjitdey94 — APPROVED** [review]

**Pull request merged**